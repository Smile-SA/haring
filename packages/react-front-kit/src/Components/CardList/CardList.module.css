.item {
  &:after {
    background: var(--mantine-color-gray-2);
    /*noinspection CssUnresolvedCustomProperty*/
    bottom: var(--separator-gap);
    content: '';
    /*noinspection CssUnresolvedCustomProperty*/
    display: var(--separator-display);
    height: 1px;
    left: 0;
    position: absolute;
    width: 100%;
  }

  &:last-child {
    &:after {
      display: none;
    }

    margin-bottom: 0;
  }

  position: relative;
}

.rootScrollBar {
  margin: 32px 0;
  padding: 0 32px;
}

.scrollBar {
  background: var(--mantine-color-gray-1);
  width: 7px !important;
}

.stack {
  display: flex;
  flex-direction: column;
  margin: 0;
}

.thumb {
  background: var(--mantine-color-gray-7);
}
