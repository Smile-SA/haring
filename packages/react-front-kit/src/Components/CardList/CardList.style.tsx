import type { MantineNumberSize } from '@mantine/styles';

import { createStyles, getSize } from '@mantine/styles';

export const useStyles = createStyles(
  (
    theme,
    { separator, spacing }: { separator: boolean; spacing: MantineNumberSize },
  ) => ({
    item: {
      '&:after': {
        background: theme.colors.gray[2],
        bottom: `calc(-${getSize({
          size: spacing,
          sizes: theme.spacing,
        })} / 2)`,
        content: '""',
        display: separator ? 'block' : 'none',
        height: '1px',
        left: '0px',
        position: 'absolute',
        width: '100%',
      },
      '&:last-child': {
        '&:after': {
          display: 'none',
        },
        marginBottom: '0',
      },
      position: 'relative',
    },
    rootScrollBar: { margin: '32px 0', padding: '0px 32px' },
    scrollBar: {
      background: theme.colors.gray[1],
      width: '7px !important',
    },
    stack: {
      display: 'flex',
      flexDirection: 'column',
      margin: '0',
    },
    thumb: {
      background: theme.colors.gray[7],
    },
  }),
);
