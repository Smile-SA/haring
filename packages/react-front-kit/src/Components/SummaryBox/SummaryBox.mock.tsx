import type { ReactElement } from 'react';

import {
  ActionIcon,
  Avatar,
  Button,
  Flex,
  Indicator,
  Text,
} from '@mantine/core';
import { Export, Eye, UserCircle } from '@phosphor-icons/react';

import { ActionRowOverflow } from '../ActionRowOverflow/ActionRowOverflow';
import {
  actionRowOverflowActionsMock,
  actionRowOverflowSelectedMock,
} from '../ActionRowOverflow/ActionRowOverflow.mock';

export const defaultSummaryBoxMock = {
  children: 'children node',
  leftNode: 'left node',
  rightNode: 'right node',
  titleNode: 'title node',
  topNode: 'top node',
};

export const topMock = (
  <span style={{ color: '#5C5F66', fontSize: 12 }}>30th novembre 2023</span>
);

export const otherTopMock = (
  <span style={{ color: '#5C5F66', fontSize: 12 }}>3 hours ago</span>
);

export const titleMock = <b>Lorem ipsum dolor sit amet</b>;

export const otherTitleMock = (
  <span>
    <b>Lorem ipsum</b>
    <span>
      dolor sit amet consectetur. Sollicitudin mattis blandit aliquet odio urna
      mi id :
    </span>
    <Text c="blue" component="span">
      Lorem Ipsum dolor
    </Text>
  </span>
);

export const childrenExampleMock = (key?: number): ReactElement => (
  <Flex key={key} direction="column" gap={24}>
    <span>
      Lorem ipsum dolor sit amet consectetur. Sollicitudin mattis blandit
      aliquet odio urna mi id. Mauris venenatis ut et at amet vel est.
    </span>
    <Flex justify="space-between" style={{ width: '100%' }}>
      <Button color="green" variant="light">
        Approved
      </Button>
      <Flex align="center" gap={16}>
        <ActionIcon color="cyan" size={30} variant="subtle">
          <Export size={24} />
        </ActionIcon>
        <ActionIcon color="cyan" size={30} variant="subtle">
          <Eye size={24} />
        </ActionIcon>
      </Flex>
    </Flex>
  </Flex>
);

export const otherChildrenExampleMock = (
  <Flex direction="column" gap={24}>
    <span>
      Lorem ipsum dolor sit amet consectetur. Sollicitudin mattis blandit
      aliquet odio urna mi id.
    </span>
    <Flex gap={5} justify="start" style={{ width: '100%' }} wrap="wrap">
      <Button color="green" variant="light">
        Approved
      </Button>
      <Button color="orange" variant="light">
        In Progress
      </Button>
      <Button color="white" variant="light">
        Paused
      </Button>
      <Button color="red" variant="light">
        Blocked
      </Button>
      <Button color="black" variant="default">
        Rejected
      </Button>
    </Flex>
  </Flex>
);

export const rightMock = (
  <ActionIcon color="cyan" size={30} variant="subtle">
    <Export size={24} />
  </ActionIcon>
);

export const actionRightMock = (
  <ActionRowOverflow
    actions={actionRowOverflowActionsMock}
    isCompactStyle
    rowActionNumber={1}
    selectedElements={actionRowOverflowSelectedMock}
  />
);

export const leftMock = (
  <Indicator color="red" inline offset={8} withBorder>
    <Avatar size={40}>
      <UserCircle size={40} />
    </Avatar>
  </Indicator>
);
