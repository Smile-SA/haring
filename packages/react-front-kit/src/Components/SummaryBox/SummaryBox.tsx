'use client';

import type { FlexProps, GroupProps, StackProps } from '@mantine/core';
import type { ReactElement, ReactNode } from 'react';

import { Flex, Group, Stack } from '@mantine/core';

import classes from './SummaryBox.module.css';

export interface ISummaryBoxProps extends FlexProps {
  children?: ReactNode;
  childrenGroupProps?: GroupProps;
  contentContainerFlexProps?: FlexProps;
  leftNode?: ReactNode;
  leftNodeStackProps?: StackProps;
  rightNode?: ReactNode;
  rightNodeStackProps?: StackProps;
  titleNode?: ReactNode;
  titleNodeGroupProps?: GroupProps;
  topNode?: ReactNode;
  topNodeGroupProps?: GroupProps;
}

export function SummaryBox(props: ISummaryBoxProps): ReactElement {
  const {
    children,
    childrenGroupProps,
    contentContainerFlexProps,
    leftNode,
    leftNodeStackProps,
    rightNode,
    rightNodeStackProps,
    titleNode,
    titleNodeGroupProps,
    topNode,
    topNodeGroupProps,
    ...containerFlexProps
  } = props;

  return (
    <Flex
      align="stretch"
      gap="md"
      justify="stretch"
      wrap="nowrap"
      {...containerFlexProps}
    >
      {Boolean(leftNode) && (
        <Stack align="stretch" justify="flex-start" {...leftNodeStackProps}>
          {leftNode}
        </Stack>
      )}
      {Boolean(topNode || titleNode || children) && (
        <Flex
          align="stretch"
          className={classes.grow}
          direction="column"
          gap={4}
          justify="start"
          {...contentContainerFlexProps}
        >
          {Boolean(topNode) && <Group {...topNodeGroupProps}>{topNode}</Group>}
          {Boolean(titleNode) && (
            <Group {...titleNodeGroupProps}>{titleNode}</Group>
          )}
          {Boolean(children) && (
            <Group className={classes.grow} {...childrenGroupProps}>
              {children}
            </Group>
          )}
        </Flex>
      )}
      {Boolean(rightNode) && (
        <Stack align="stretch" justify="flex-start" {...rightNodeStackProps}>
          {rightNode}
        </Stack>
      )}
    </Flex>
  );
}
