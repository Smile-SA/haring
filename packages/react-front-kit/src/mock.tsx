/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
