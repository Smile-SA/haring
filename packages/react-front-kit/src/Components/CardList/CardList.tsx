'use client';

import type { MantineNumberSize, ScrollAreaProps } from '@mantine/core';
import type { ReactElement, ReactNode } from 'react';

import { ScrollArea, Stack } from '@mantine/core';

import { useStyles } from './CardList.style';

export interface ICardListProps extends Omit<ScrollAreaProps, 'children'> {
  children: ReactElement | ReactNode[];
  separator?: boolean;
  spacing?: MantineNumberSize;
}

export function CardList(props: ICardListProps): ReactElement {
  const {
    children,
    separator = true,
    spacing = 'xl',
    ...scrollAreaProps
  } = props;
  const { classes } = useStyles({ separator, spacing });
  return (
    <ScrollArea
      {...scrollAreaProps}
      classNames={{
        root: classes.rootScrollBar,
        scrollbar: classes.scrollBar,
        thumb: classes.thumb,
      }}
    >
      {Array.isArray(children) ? (
        <Stack className={classes.stack} spacing={spacing}>
          {children.map((item: ReactNode, index: number) => (
            <div key={`${index + index}`} className={classes.item}>
              {item}
            </div>
          ))}
        </Stack>
      ) : (
        <div className={classes.item}>{children}</div>
      )}
    </ScrollArea>
  );
}
