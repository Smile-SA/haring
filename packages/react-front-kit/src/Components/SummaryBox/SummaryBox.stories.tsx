import type { Meta, StoryObj } from '@storybook/react';
import type { ReactElement } from 'react';

import { Container } from '@mantine/core';

import { SummaryBox as Cmp } from './SummaryBox';
import {
  childrenExampleMock,
  defaultSummaryBoxMock,
  leftMock,
  otherTitleMock,
  otherTopMock,
  rightMock,
  titleMock,
  topMock,
} from './SummaryBox.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/SummaryBox',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const SummaryBox: IStory = {
  args: defaultSummaryBoxMock,
};

const container = ({ ...props }): ReactElement => (
  <Container bg="white" size="400px">
    <Cmp {...props} />
  </Container>
);

export const ChildrenExample: IStory = {
  args: {
    children: childrenExampleMock,
    titleNode: titleMock,
    topNode: topMock,
    topNodeGroupProps: { mb: 12 },
  },
  render: ({ ...props }) => container(props),
};

export const SimpleRightNodeExample: IStory = {
  args: {
    pb: 24,
    rightNode: rightMock,
    titleNode: titleMock,
  },
  render: ({ ...props }) => container(props),
};

export const LeftNodeExample: IStory = {
  args: {
    leftNode: leftMock,
    titleNode: otherTitleMock,
    topNode: otherTopMock,
  },
  render: ({ ...props }) => container(props),
};
