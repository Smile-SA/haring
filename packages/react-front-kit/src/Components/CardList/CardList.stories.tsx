import type { Meta, StoryObj } from '@storybook/react';

import { Card } from '@mantine/core';

import { CardHeader } from '../CardHeader/CardHeader';
import { childrenExampleMock } from '../SummaryBox/SummaryBox.mock';

import { CardList as Cmp } from './CardList';

const summaryboxs = [
  childrenExampleMock,
  childrenExampleMock,
  childrenExampleMock,
  childrenExampleMock,
  childrenExampleMock,
];

const texts = ['Hello-world', 'Hello-world', 'Hello-world'];

const meta = {
  argTypes: {
    spacing: {
      control: { type: 'select' },
      options: ['xl', 'lg', 'md', 'sm', 'xs'],
    },
  },
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/CardList',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const CardListWithMultiSummaryBox: IStory = {
  args: {
    children: summaryboxs,
    h: '200px',
    separator: true,
    spacing: 'xl',
  },
};

export const CardListWithOneSummaryBox: IStory = {
  args: {
    children: childrenExampleMock,
    h: '110px',
    separator: true,
    spacing: 'xl',
  },
};

export const CardListWithTexts: IStory = {
  args: {
    children: texts,
    h: '70px',
    spacing: 'xl',
  },
  render: ({ ...props }) => (
    <Card>
      <Card.Section>
        <CardHeader>
          <h1 style={{ margin: '24px', marginBottom: 0 }}>Titre</h1>
        </CardHeader>
        <Cmp {...props} />
      </Card.Section>
    </Card>
  ),
};
