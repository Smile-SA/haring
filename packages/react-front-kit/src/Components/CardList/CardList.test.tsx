import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { childrenExampleMock } from '../SummaryBox/SummaryBox.mock';

import { CardList } from './CardList';

describe('CardList', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <CardList>
        {[
          childrenExampleMock,
          childrenExampleMock,
          childrenExampleMock,
          childrenExampleMock,
        ]}
      </CardList>,
    );
    expect(container).toMatchSnapshot();
  });
});
