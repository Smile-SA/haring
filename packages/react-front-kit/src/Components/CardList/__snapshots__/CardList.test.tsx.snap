// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList matches snapshot 1`] = `
<div>
  <div
    class="mantine-ScrollArea-root mantine-e9uey"
    dir="ltr"
    style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
  >
    <style>
      [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
    </style>
    <div
      class="mantine-b6zkvl mantine-ScrollArea-viewport"
      data-radix-scroll-area-viewport=""
      style="overflow-x: scroll; overflow-y: scroll;"
    >
      <div
        style="min-width: 100%; display: table;"
      >
        <div
          class="mantine-Stack-root mantine-1ute00k"
        >
          <div
            class="mantine-1hf096k"
          >
            <div
              class="mantine-1pure0w"
            >
              <span>
                Lorem ipsum dolor sit amet consectetur. Sollicitudin mattis blandit aliquet odio urna mi id. Mauris venenatis ut et at amet vel est.
              </span>
              <div
                class="mantine-73o8aw"
                style="width: 100%;"
              >
                <button
                  class="mantine-UnstyledButton-root mantine-Button-root mantine-72wjox"
                  data-button="true"
                  type="button"
                >
                  <div
                    class="mantine-1wpc1xj mantine-Button-inner"
                  >
                    <span
                      class="mantine-1ryt1ht mantine-Button-label"
                    >
                      Approved
                    </span>
                  </div>
                </button>
                <div
                  class="mantine-1qizhnu"
                >
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-tos26d"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="20"
                      viewBox="0 0 256 256"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M216,112v96a16,16,0,0,1-16,16H56a16,16,0,0,1-16-16V112A16,16,0,0,1,56,96H80a8,8,0,0,1,0,16H56v96H200V112H176a8,8,0,0,1,0-16h24A16,16,0,0,1,216,112ZM93.66,69.66,120,43.31V136a8,8,0,0,0,16,0V43.31l26.34,26.35a8,8,0,0,0,11.32-11.32l-40-40a8,8,0,0,0-11.32,0l-40,40A8,8,0,0,0,93.66,69.66Z"
                      />
                    </svg>
                  </button>
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-tos26d"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="20"
                      viewBox="0 0 256 256"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M247.31,124.76c-.35-.79-8.82-19.58-27.65-38.41C194.57,61.26,162.88,48,128,48S61.43,61.26,36.34,86.35C17.51,105.18,9,124,8.69,124.76a8,8,0,0,0,0,6.5c.35.79,8.82,19.57,27.65,38.4C61.43,194.74,93.12,208,128,208s66.57-13.26,91.66-38.34c18.83-18.83,27.3-37.61,27.65-38.4A8,8,0,0,0,247.31,124.76ZM128,192c-30.78,0-57.67-11.19-79.93-33.25A133.47,133.47,0,0,1,25,128,133.33,133.33,0,0,1,48.07,97.25C70.33,75.19,97.22,64,128,64s57.67,11.19,79.93,33.25A133.46,133.46,0,0,1,231.05,128C223.84,141.46,192.43,192,128,192Zm0-112a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160Z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="mantine-1hf096k"
          >
            <div
              class="mantine-1pure0w"
            >
              <span>
                Lorem ipsum dolor sit amet consectetur. Sollicitudin mattis blandit aliquet odio urna mi id. Mauris venenatis ut et at amet vel est.
              </span>
              <div
                class="mantine-73o8aw"
                style="width: 100%;"
              >
                <button
                  class="mantine-UnstyledButton-root mantine-Button-root mantine-72wjox"
                  data-button="true"
                  type="button"
                >
                  <div
                    class="mantine-1wpc1xj mantine-Button-inner"
                  >
                    <span
                      class="mantine-1ryt1ht mantine-Button-label"
                    >
                      Approved
                    </span>
                  </div>
                </button>
                <div
                  class="mantine-1qizhnu"
                >
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-tos26d"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="20"
                      viewBox="0 0 256 256"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M216,112v96a16,16,0,0,1-16,16H56a16,16,0,0,1-16-16V112A16,16,0,0,1,56,96H80a8,8,0,0,1,0,16H56v96H200V112H176a8,8,0,0,1,0-16h24A16,16,0,0,1,216,112ZM93.66,69.66,120,43.31V136a8,8,0,0,0,16,0V43.31l26.34,26.35a8,8,0,0,0,11.32-11.32l-40-40a8,8,0,0,0-11.32,0l-40,40A8,8,0,0,0,93.66,69.66Z"
                      />
                    </svg>
                  </button>
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-tos26d"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="20"
                      viewBox="0 0 256 256"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M247.31,124.76c-.35-.79-8.82-19.58-27.65-38.41C194.57,61.26,162.88,48,128,48S61.43,61.26,36.34,86.35C17.51,105.18,9,124,8.69,124.76a8,8,0,0,0,0,6.5c.35.79,8.82,19.57,27.65,38.4C61.43,194.74,93.12,208,128,208s66.57-13.26,91.66-38.34c18.83-18.83,27.3-37.61,27.65-38.4A8,8,0,0,0,247.31,124.76ZM128,192c-30.78,0-57.67-11.19-79.93-33.25A133.47,133.47,0,0,1,25,128,133.33,133.33,0,0,1,48.07,97.25C70.33,75.19,97.22,64,128,64s57.67,11.19,79.93,33.25A133.46,133.46,0,0,1,231.05,128C223.84,141.46,192.43,192,128,192Zm0-112a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160Z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="mantine-1hf096k"
          >
            <div
              class="mantine-1pure0w"
            >
              <span>
                Lorem ipsum dolor sit amet consectetur. Sollicitudin mattis blandit aliquet odio urna mi id. Mauris venenatis ut et at amet vel est.
              </span>
              <div
                class="mantine-73o8aw"
                style="width: 100%;"
              >
                <button
                  class="mantine-UnstyledButton-root mantine-Button-root mantine-72wjox"
                  data-button="true"
                  type="button"
                >
                  <div
                    class="mantine-1wpc1xj mantine-Button-inner"
                  >
                    <span
                      class="mantine-1ryt1ht mantine-Button-label"
                    >
                      Approved
                    </span>
                  </div>
                </button>
                <div
                  class="mantine-1qizhnu"
                >
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-tos26d"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="20"
                      viewBox="0 0 256 256"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M216,112v96a16,16,0,0,1-16,16H56a16,16,0,0,1-16-16V112A16,16,0,0,1,56,96H80a8,8,0,0,1,0,16H56v96H200V112H176a8,8,0,0,1,0-16h24A16,16,0,0,1,216,112ZM93.66,69.66,120,43.31V136a8,8,0,0,0,16,0V43.31l26.34,26.35a8,8,0,0,0,11.32-11.32l-40-40a8,8,0,0,0-11.32,0l-40,40A8,8,0,0,0,93.66,69.66Z"
                      />
                    </svg>
                  </button>
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-tos26d"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="20"
                      viewBox="0 0 256 256"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M247.31,124.76c-.35-.79-8.82-19.58-27.65-38.41C194.57,61.26,162.88,48,128,48S61.43,61.26,36.34,86.35C17.51,105.18,9,124,8.69,124.76a8,8,0,0,0,0,6.5c.35.79,8.82,19.57,27.65,38.4C61.43,194.74,93.12,208,128,208s66.57-13.26,91.66-38.34c18.83-18.83,27.3-37.61,27.65-38.4A8,8,0,0,0,247.31,124.76ZM128,192c-30.78,0-57.67-11.19-79.93-33.25A133.47,133.47,0,0,1,25,128,133.33,133.33,0,0,1,48.07,97.25C70.33,75.19,97.22,64,128,64s57.67,11.19,79.93,33.25A133.46,133.46,0,0,1,231.05,128C223.84,141.46,192.43,192,128,192Zm0-112a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160Z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="mantine-1hf096k"
          >
            <div
              class="mantine-1pure0w"
            >
              <span>
                Lorem ipsum dolor sit amet consectetur. Sollicitudin mattis blandit aliquet odio urna mi id. Mauris venenatis ut et at amet vel est.
              </span>
              <div
                class="mantine-73o8aw"
                style="width: 100%;"
              >
                <button
                  class="mantine-UnstyledButton-root mantine-Button-root mantine-72wjox"
                  data-button="true"
                  type="button"
                >
                  <div
                    class="mantine-1wpc1xj mantine-Button-inner"
                  >
                    <span
                      class="mantine-1ryt1ht mantine-Button-label"
                    >
                      Approved
                    </span>
                  </div>
                </button>
                <div
                  class="mantine-1qizhnu"
                >
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-tos26d"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="20"
                      viewBox="0 0 256 256"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M216,112v96a16,16,0,0,1-16,16H56a16,16,0,0,1-16-16V112A16,16,0,0,1,56,96H80a8,8,0,0,1,0,16H56v96H200V112H176a8,8,0,0,1,0-16h24A16,16,0,0,1,216,112ZM93.66,69.66,120,43.31V136a8,8,0,0,0,16,0V43.31l26.34,26.35a8,8,0,0,0,11.32-11.32l-40-40a8,8,0,0,0-11.32,0l-40,40A8,8,0,0,0,93.66,69.66Z"
                      />
                    </svg>
                  </button>
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-tos26d"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="20"
                      viewBox="0 0 256 256"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M247.31,124.76c-.35-.79-8.82-19.58-27.65-38.41C194.57,61.26,162.88,48,128,48S61.43,61.26,36.34,86.35C17.51,105.18,9,124,8.69,124.76a8,8,0,0,0,0,6.5c.35.79,8.82,19.57,27.65,38.4C61.43,194.74,93.12,208,128,208s66.57-13.26,91.66-38.34c18.83-18.83,27.3-37.61,27.65-38.4A8,8,0,0,0,247.31,124.76ZM128,192c-30.78,0-57.67-11.19-79.93-33.25A133.47,133.47,0,0,1,25,128,133.33,133.33,0,0,1,48.07,97.25C70.33,75.19,97.22,64,128,64s57.67,11.19,79.93,33.25A133.46,133.46,0,0,1,231.05,128C223.84,141.46,192.43,192,128,192Zm0-112a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160Z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
